homeassistant/components/nut/__init__.py

```diff
@@ -83,10 +83,18 @@ async def async_update_data():
 
     undo_listener = entry.add_update_listener(_async_update_listener)
 
+    unique_id = _unique_id_from_status(status)
+
+    if unique_id is None:
+        # If the device is missing a serial number
+        # we fallback to the host/port/alias
+        # as the sensors won't show up on the integrations page.
+        unique_id = format_host_port_alias(config)
```

Comment:
We should not base unique ID's on that, as they should not be able to change.

<https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements>

Reply:
@frenck Any ideas on a solution from reading this issue https://github.com/home-assistant/core/issues/33908 ?

Reply:
The unique ID should not be set in those cases.

Reply:
Maybe use the host MAC address if the serial number is not available?

Reply:
@bieniu I used the mac address when the serial is unavailable.